fix(formatter): JSX text wrapping incorrect 2

Dunqing opened the pull request:
* close: #16199

~~Two tests regressed, but I kind of think they are correct. I created https://github.com/prettier/prettier/issues/18384 to confirm the issue in Prettier.~~ Aligned with Prettier's current behavior

Dunqing commented:
Finally, I figured out why it produces different output while the IR/Doc is basically the same. This is due to the slight differences in handling `measured_group_fits` in the Oxfmt compared to Prettier, which is called as [shouldRemeasure](https://github.com/prettier/prettier/blob/0273e33fc691e28e4ab3f3c8ee86918b65cf823d/src/document/printer/printer.js#L174-L175) in Prettier and represents the opposite meaning. 

After exactly matching what `shouldRemeasure` does in Prettier, two regressed cases are gone.

leaysgur reviewed (approved):
Also confirmed all `{' '}` related diffs in next.js repo are gone... 👍🏻

leaysgur commented:
### Merge activity

* **Dec 3, 6:58 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 3, 6:59 AM UTC**: `leaysgur` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Dunqing committed:
fix(formatter): JSX text wrapping incorrect 2 (#16320)

* close: #16199

~~Two tests regressed, but I kind of think they are correct. I created https://github.com/prettier/prettier/issues/18384 to confirm the issue in Prettier.~~ Aligned with Prettier's current behavior
